### PR TITLE
Implement notification client

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Create scheduled job for processing notifications](https://tools.hmcts.net/jira/browse/BPS-288)

### Change description ###

This fully implements client. Since nothing ties to it as of yet, just including smoke tests as a "probe" thing verifying client works as expected

Note, PR is based from #337 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/smokeTest/java/uk/gov/hmcts/reform/bulkscanprocessor/ErrorNotificationClientTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    /*
+    NOTICE.
+    These are smoke tests so will NOT include any happy scenarios.
+    They will be covered with functional tests once everything is plugged in.
+     */
+
+    @Test
+    public void should_return_NotificationClientException_with_multiple_missing_required_fields_info() {
+        Throwable throwable = catchThrowable(() ->
+            System.out.println(client.notify(new ErrorNotificationRequest(
```
Comment (rkondratowicz) [inline src/smokeTest/java/uk/gov/hmcts/reform/bulkscanprocessor/ErrorNotificationClientTest.java]: this is NOT a smoke test of the service

Comment (doncem) [reply src/smokeTest/java/uk/gov/hmcts/reform/bulkscanprocessor/ErrorNotificationClientTest.java]: yep. will move to functional

**Comment by doncem** [comment]
I'll split client related stuff and squeeze into #337 so at least decoder part can be unit tested

**Comment by doncem** [comment]
lost it's purpose. will create portion of this as a separate pr

**Pull request closed without merging**